addons/AsepriteWizard/import_plugin.gd

```diff
@@ -0,0 +1,232 @@
+	dir.open(save_path)
+	dir.list_dir_begin()
+	var file_name = dir.get_next()
+	while file_name != "":
+		if file_name != '.' and file_name != '..':
+			dir.remove(file_name)
+		file_name = dir.get_next()
+
+	var export_mode = aseprite.LAYERS_EXPORT_MODE if options['split_layers'] else aseprite.FILE_EXPORT_MODE
+
+	var aseprite_opts = {
+		"export_mode": export_mode,
+		"exception_pattern": options['exclude_layers_pattern'],
+		"only_visible_layers": options['only_visible_layers'],
+		"trim_images": options['trim_images'],
+		"output_filename": ''
+	}
+
+	var exit_code = aseprite.create_resource(absolute_source_file, absolute_save_path, aseprite_opts)
+	if exit_code != 0:
+		print("ERROR - Could not import aseprite file: %s" % get_error_message(exit_code))
+		return FAILED
+
+	dir.open(save_path)
+	dir.list_dir_begin()
+
+	file_name = dir.get_next()
+
+	var main_sprite_frame_saved = false
+
+	var global_replacement_vars = {
+		"basename": source_basename,
+	}
+
+	# Scan through the import directory and process the generated resources based on what options have been selected.
+	while file_name != "":
+		if file_name != ".." and file_name != ".":
+			if file_name.ends_with(".res"):
+				# This is a SpriteFrames resource generated for a layer.
+				var local_replacement_vars = global_replacement_vars.duplicate()
+				local_replacement_vars["layer"] = file_name.substr(0, file_name.length() - 4)
+
+				if not main_sprite_frame_saved:
+					# Save this resource as the main resource. We need to set something here or Godot won't stop
+					# re-importing the resource. So this is either the SpriteFrames instance of the first layer
+					# (alphabetically) found in the import directory.
+					var sprite_frames : SpriteFrames = ResourceLoader.load("%s/%s" % [save_path, file_name], 'SpriteFrames', true)
+					main_sprite_frame_saved = true
+					var resource_path = "%s.res" % save_path;
+					sprite_frames.take_over_path(resource_path)
+					ResourceSaver.save(resource_path, sprite_frames)
+
+				var sprite_frames : SpriteFrames = ResourceLoader.load("%s/%s" % [save_path, file_name], 'SpriteFrames', true)
+
+				if options["split_layers"]:
+					var sprite_replacement_vars = local_replacement_vars.duplicate()
+					sprite_replacement_vars["extension"] = "res"
+
+					var sprite_filename = "%s/%s" % [source_path, replace_vars(options["sprite_filename_pattern"], sprite_replacement_vars)]
+					ResourceSaver.save(sprite_filename, sprite_frames)
+					sprite_frames.take_over_path(sprite_filename)
+
+				if options["texture_atlas/import_texture_atlas"]:
+					# Create a TextureAtlas resource for this layer.
+					var atlas_texture = null
+					var replacement_vars = local_replacement_vars.duplicate()
+
+					for anim in sprite_frames.animations:
+						var i=0
+
+						replacement_vars["animation"] = anim.name
+						replacement_vars["extension"] = "res"
+
+						for frame in anim.frames:
+							replacement_vars["frame"] = i
+
+							if not atlas_texture:
```

Comment:
Can you please comment on this section?

Reply:
This is an external plugin. I don't know what this code does